main/acle.md

```diff
@@ -9017,74 +9021,71 @@ The intrinsics in this section have the following properties in common:
 ``` c
   // Also for _za16, _za32, _za64 and _za128 (with the same prototype).
   __attribute__((arm_streaming, arm_shared_za))
-  void svld1_hor_za8(uint64_t tile, uint32_t slice_base,
-                     uint64_t slice_offset, svbool_t pg, const void *ptr);
+  void svld1_hor_za8(uint64_t tile, uint32_t slice, svbool_t pg,
+                     const void *ptr);
 
   // Synthetic intrinsic: adds vnum * svcntsb() to the address given by ptr.
   // Also for _za16, _za32, _za64 and _za128 (with the same prototype).
   __attribute__((arm_streaming, arm_shared_za))
-  void svld1_hor_vnum_za8(uint64_t tile, uint32_t slice_base,
-                          uint64_t slice_offset, svbool_t pg,
+  void svld1_hor_vnum_za8(uint64_t tile, uint32_t slice, svbool_t pg,
                           const void *ptr, int64_t vnum);
 
   // Also for _za16, _za32, _za64 and _za128 (with the same prototype).
   __attribute__((arm_streaming, arm_shared_za))
-  void svld1_ver_za8(uint64_t tile, uint32_t slice_base,
-                     uint64_t slice_offset, svbool_t pg, const void *ptr);
+  void svld1_ver_za8(uint64_t tile, uint32_t slice, svbool_t pg,
+                     const void *ptr);
 
   // Synthetic intrinsic: adds vnum * svcntsb() to the address given by ptr.
   // Also for _za16, _za32, _za64 and _za128 (with the same prototype).
   __attribute__((arm_streaming, arm_shared_za))
-  void svld1_ver_vnum_za8(uint64_t tile, uint32_t slice_base,
-                          uint64_t slice_offset, svbool_t pg,
+  void svld1_ver_vnum_za8(uint64_t tile, uint32_t slice, svbool_t pg,
                           const void *ptr, int64_t vnum);
 ```
 
 #### LDR
 
 ``` c
-  // slice_offset fills the role of the usual vnum parameter.
   __attribute__((arm_streaming_compatible, arm_shared_za))
-  void svldr_vnum_za(uint32_t slice_base, uint64_t slice_offset,
-                     const void *ptr);
+  void svldr_za(uint32_t slice, const void *ptr);
+
+  __attribute__((arm_streaming_compatible, arm_shared_za))
+  void svldr_vnum_za(uint32_t slice, const void *ptr, int64_t vnum);
```

Comment:
With slice and vnum being independent values, I guess this means that vnum and slice offset can be different, such that:
```
  svldr_vnum_za(1, ptr, 1)
=> 
  mov w12, #0
  ldr za[w12, 1], [x0, #1, mul vl]
```
and
```
  svldr_vnum_za(1, ptr, 2)
=> 
  mov w12, #0
  addsvl x0, #1
  ldr za[w12, 1], [x0, #1, mul vl]

or
  mov w12, #-1
  ldr za[w12, 2], [x0, #2, mul vl]
```
and
```
  svldr_vnum_za(tile, ptr, 2)
=>
  sub w12, #2
  ldr za[w12, 2], [x0, #2, mul vl]
```

The extra instructions to match the immediates may be unexpected by people using this intrinsic. is it worth either keeping the svldr instruction in the old form, or otherwise pointing out how to get the best results from this intrinsic in a comment? (e.g. `svldr_num_za(slice + 2, ptr, 2)` can be done in one instruction, whereas `svldr_vnum_za(slice, ptr, 2)` cannot)